snuba/subscriptions/store.py

```diff
@@ -1,76 +1,48 @@
-import json
-from datetime import timedelta
+from uuid import UUID
 from typing import Collection, Tuple
 
 from snuba.datasets.dataset import Dataset
 from snuba.datasets.factory import get_dataset_name
 from snuba.redis import RedisClientType
-from snuba.subscriptions.data import Subscription
-from snuba.utils.codecs import Codec
+from snuba.subscriptions.codecs import SubscriptionDataCodec
+from snuba.subscriptions.data import PartitionId, SubscriptionData
 
 
-class SubscriptionCodec(Codec[bytes, Subscription]):
-    def encode(self, value: Subscription) -> bytes:
-        return json.dumps(
-            {
-                "project_id": value.project_id,
-                "conditions": value.conditions,
-                "aggregations": value.aggregations,
-                "time_window": int(value.time_window.total_seconds()),
-                "resolution": int(value.resolution.total_seconds()),
-            }
-        ).encode("utf-8")
-
-    def decode(self, value: bytes) -> Subscription:
-        data = json.loads(value.decode("utf-8"))
-        return Subscription(
-            project_id=data["project_id"],
-            conditions=data["conditions"],
-            aggregations=data["aggregations"],
-            time_window=timedelta(seconds=data["time_window"]),
-            resolution=timedelta(seconds=data["resolution"]),
-        )
-
-
-class RedisSubscriptionStore:
+class RedisSubscriptionDataStore:
```

Comment:
It might be worth abstracting this out into a class that deals with Redis mappings with key/value encoders.